observability/micrometer/src/main/java/org/wildfly/extension/micrometer/MicrometerSubsystemAdd.java

```diff
@@ -57,40 +55,37 @@ public class MicrometerSubsystemAdd extends AbstractBoottimeAddStepHandler {
     @Override
     protected void performBoottime(OperationContext context, ModelNode operation, ModelNode model)
             throws OperationFailedException {
-        // If the MP Metrics module is not installed, we need to install the WF Metrics DPU and initiate a metrics
-        // collection. If MP Metrics *is* installed, then we do not need to do either of those things, as that module
-        // handles that instead.
-        if (!context.getCapabilityServiceSupport().hasCapability(METRICS_SCAN_CAPABILITY)) {
-            List<String> exposedSubsystems = MicrometerSubsystemDefinition.EXPOSED_SUBSYSTEMS.unwrap(context, model);
-            boolean exposeAnySubsystem = exposedSubsystems.remove("*");
-            boolean securityEnabled = MicrometerSubsystemDefinition.SECURITY_ENABLED.resolveModelAttribute(context, model)
-                    .asBoolean();
 
-            MicrometerRegistryService.install(context, securityEnabled);
-            MicrometerCollectorService.install(context);
-            MicrometerContextService.install(context, securityEnabled);
+        List<String> exposedSubsystems = MicrometerSubsystemDefinition.EXPOSED_SUBSYSTEMS.unwrap(context, model);
+        boolean exposeAnySubsystem = exposedSubsystems.remove("*");
+        boolean securityEnabled = MicrometerSubsystemDefinition.SECURITY_ENABLED.resolveModelAttribute(context, model)
+                .asBoolean();
+
+        MicrometerRegistryService.install(context, securityEnabled);
+        MicrometerCollectorService.install(context);
+        MicrometerContextService.install(context, securityEnabled);
+
+        context.addStep(new AbstractDeploymentChainStep() {
+            @Override
+            public void execute(DeploymentProcessorTarget processorTarget) {
+                processorTarget.addDeploymentProcessor(SUBSYSTEM_NAME, DEPENDENCIES, 0x1910,
+                        new MicrometerDependencyProcessor());
+                processorTarget.addDeploymentProcessor(SUBSYSTEM_NAME, POST_MODULE, POST_MODULE_METRICS - 1, // ???
```

Comment:
IIRC you added a Phase constant for this in WF Core, which will replace this when integrated in full WF. Is this the same value as what that will be?

Reply:
Yes, that constant will replace this, but it doesn't look like Core has been updated yet, so I can't use the constant.